frontend/components/dashboard/new-session.tsx

```diff
@@ -25,24 +27,92 @@ export const NewSession = () => {
     const [selectedTopic, setSelectedTopic] = React.useState('')
     const [selectedComplexity, setSelectedComplexity] = React.useState('')
 
+    const [modalData, setModalData] = React.useState({
+        isOpen: false,
+        isMatchmaking: false,
+        isMatchFound: false,
+    })
+    const [timeElapsed, setTimeElapsed] = React.useState({
```

Comment:
it's fine to just track time by seconds, then can reduce a lot of logic and less variables also 👍